expose-app-api/routes/user.js

```diff
@@ -0,0 +1,106 @@
+const express = require ('express');
+const router = express.Router();
+const bcrypt = require('bcrypt');
+const jwt = require ('jsonwebtoken');
+const user = require ('../models/users');
+
+
+// user registration 
+router.post('/register', async(req, res)=> { 
+    try {
+
+        const { username, email, password} = req.body;
+
+        // now checking of user already exists.
+        const existingUser = await user.findOne({where: {email}|| {username}});
```

Comment:
how does this work, does findOne use the email or the username?

Reply:
my initial thought was to use only email as the unique data to avoid one email creating more accounts but later decoded to have one unique username to a user to avoid duplication of user name. so that lines are checking if there is an already existing  user with the same username or email

Reply:
yeah the "product" makes sense, but how does the findOne work?

we can test this once you have the UI making calls to this end point!